src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -90,29 +90,29 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
 
         # Do stuff with DBS
         try:
+            # check if block exists in DBS
+            results = dbsApi.listBlocks(block_name=name)
+            # the results are shown in the following form
+            # [{'block_name': '/block/name#123'}]
+            if len(results) == 1 and results[0]['block_name'] == name:
+                # found that we have this block, i.e. no need to insert block anymore
+                logging.info("block %s already exist in DBS", block)
+                continue
             logging.info("About to call insert block for: %s", name)
             dbsApi.insertBulkBlock(blockDump=block)
             results.put({'name': name, 'success': "uploaded"})
         except HTTPError as ex:
             # DBS Go server errors are defined here:
             # https://github.com/dmwm/dbs2go/blob/master/dbs/errors.go
             dbsError = DBSError(ex.body)
-            reason = dbsError.getReason()
             message = dbsError.getMessage()
             srvCode = dbsError.getServerCode()
-            msg = f'DBSError code: {srvCode}, message: {message}, reason: {reason}'
-            if srvCode == 128:
-                # block already exist
-                logging.warning("Block %s already exists. Marking it as uploaded.", name)
-                results.put({'name': name, 'success': "check"})
-            elif srvCode in [132, 133, 134, 135, 136, 137, 138, 139, 140]:
-                # racing conditions
-                logging.warning("Hit a transient data race condition injecting block %s, %s", name, msg)
-                results.put({'name': name, 'success': "error", 'error': msg})
-            else:
-                msg = f"Error trying to process block {name} through DBS. Details: {msg}"
-                logging.error(msg)
-                results.put({'name': name, 'success': "error", 'error': msg})
+            errors = DBSErrors(dbsUrl)
```

Comment:
Other than the concern expressed above, how much is the output of DBSErrors supposed to change?
I believe it will be pretty much static, and if so, we could use a caching system and avoid much of the I/O.

Reply:
Once again I against over optimization where it is not obvious. This API call to DBS server do not even touch its database. The HTTP cal which `DBSErrors` do is negligible, while caching will lead to a possibility that if we upgrade dbs server and it changes code (add new code) that we'll have stale cache which someone will need to take into account. I don't see much need to introduce such cache.

Reply:
Yes, I agree that using cache always increases the complexity of the code. On the other hand, I feel like we need to adopt efficient design and implementation whenever we see fit, especially as the scale of data will only increase in the next years.

I guess a good compromisse would be to fetch these server errors once this method `uploadWorker` starts, above the `while True`. This way we ensure that we have only one call for each cycle of the component.
What do you think?

Reply:
ok, I agree , it is good suggestion to move it outside of while loop.